io-ballerina/src/io/bytes/block_stream.bal

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/java;
+
+type Block readonly & byte[];
+
+# BlockStream used to initialize a byte stream.
+public class BlockStream {
+    private ReadableByteChannel readableByteChannel;
+    private int blockSize;
+
+    public function init(ReadableByteChannel readableByteChannel, int blockSize) {
+        self.readableByteChannel = readableByteChannel;
+        self.blockSize = blockSize;
+    }
+
+    public isolated function next() returns record {| Block value; |}? {
+        var block = readBlock(self.readableByteChannel, self.blockSize);
+        if (block is byte[]) {
+            record {| Block value; |} value = {value: <Block>block.cloneReadOnly()};
+            return value;
+        } else {
+            var closeResult = closeInputStream(self.readableByteChannel);
```

Comment:
```suggestion
            _ = closeInputStream(self.readableByteChannel);
```

Reply:
This gives
```
error: ballerina/io:0.5.2::bytes/block_stream.bal:37:17: incompatible types: expected 'any', found '(ballerina/io:0.5.2:GenericError|ballerina/io:0.5.2:ConnectionTimedOutError|ballerina/io:0.5.2:AccessDeniedError|ballerina/io:0.5.2:FileNotFoundError|ballerina/io:0.5.2:TypeMismatchError|ballerina/io:0.5.2:EofError)?'
```

Reply:
Could you raise this in slack? This should ignore the result and not give typemismatcherror

Reply:
Since this is not a breaking change, we can open another issue to track this